include/stl2/detail/algorithm/sort.hpp

```diff
@@ -24,81 +24,125 @@
 // sort [sort]
 //
 STL2_OPEN_NAMESPACE {
-	template<RandomAccessIterator I, Sentinel<I> S, class Comp = less,
-		class Proj = identity>
-	requires
-		Sortable<I, Comp, Proj>
-	I sort(I first, S sent, Comp comp = {}, Proj proj = {})
-	{
-		if (first == sent) {
-			return first;
+	struct __sort_fn : private __niebloid {
+	private:
+		static constexpr std::ptrdiff_t introsort_threshold = 16;
+
+		template<RandomAccessIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr I unguarded_partition(I first, I last, Comp& comp, Proj& proj)
+		{
+			I pivot_pnt = detail::rsort::choose_pivot(first, last, comp, proj);
+
+			// Do the partition:
+			while (true) {
+				auto &&v = *pivot_pnt;
+				auto &&pivot = __stl2::invoke(proj, (decltype(v) &&)v);
+				while (__stl2::invoke(comp, __stl2::invoke(proj, *first), pivot)) {
+					++first;
+				}
+				while (__stl2::invoke(comp, pivot, __stl2::invoke(proj, *--last))) {
+					[[fallthrough]]; // there's no [[do_nothing]] attribute
+				}
+				if (!(first < last)) {
+					return first;
+				}
+				iter_swap(first, last);
+				pivot_pnt = pivot_pnt == first ? last : (pivot_pnt == last ? first : pivot_pnt);
+				++first;
+			}
+		}
+
+		template<RandomAccessIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr void introsort_loop(I first, I last, iter_difference_t<I> depth_limit,
+			Comp& comp, Proj& proj)
+		{
+			while (distance(first, last) > introsort_threshold) {
```

Comment:
`distance` isn't yet a niebloid.
```suggestion
			while (__stl2::distance(first, last) > introsort_threshold) {
```

Reply:
(Occurs repeatedly.)

Reply:
I'm tempted to fix this in a separate, please-merge-before-#225 PR.

Reply:
> I'm tempted to fix this in a separate, please-merge-before-#225 PR.

That works, too. Don't forget to make an __advance_fn::operator()` for `advance(counted_iterator)`.

Reply:
🤦‍♀️  #199 made `next`, `prev`, and `distance` niebloids back in October. Nevermind. (Don't let this stop you from implementing `advance` if you're so motivated.)